src/shared/providers/alerts/AlertsProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, ReactNode, useContext } from 'react';
+
+import { Alert } from "ui"
+
+namespace AlertsProvider {
+    export interface State {
+        text: string;
+        display: boolean;
+        alerts: { id: number, text: string, type: 'warning' | 'error' | 'success' | 'info' }[];
+        addAlert?(text: string, type: 'warning' | 'error' | 'success' | 'info'): void;
+    }
+
+    export interface Props {
+        children: ReactNode;
+    }
+}
+
+const STATE: AlertsProvider.State = {
+    text: "",
+    display: false,
+    alerts: []
+};
+
+const DURATION = 5000; //alert duration
+
+const Context = createContext(STATE);
+
+class Provider extends React.Component<AlertsProvider.Props, typeof STATE> {
+    sleep = (milliseconds) => {
+        return new Promise(resolve => setTimeout(resolve, milliseconds))
+    }
+
+    addAlert = async (text: string, type: 'warning' | 'error' | 'success' | 'info') => {
+        let id = 0;
```

Comment:
Please add some white space to make this code prettier. Also format this file - stop using one line ifs

Reply:
done